### PR TITLE
add onnxruntime custom op's backend

### PR DESCRIPTION
**Pull request by tangyanf**
I moved the previous pr to this

### CODE REVIEW

File: onnxsim/__main__.py [hunk 1]
```diff
@@ -43,12 +49,33 @@ def main():
                 # for the input name like input:0
                 name, shape = ':'.join(
                     pieces[:-1]), list(map(int, pieces[-1].split(',')))
-                input_shapes[name] = shape
+                input_shapes.update({name: shape})
+
+    input_datas = dict()
```
Comment (daquexian) [inline onnxsim/__main__.py:54]: `input_data_paths` is a better name

File: onnxsim/__main__.py [hunk 2]
```diff
@@ -43,12 +49,33 @@ def main():
...
+    if args.input_data is not None:
+        for x in args.input_data:
+            pieces = x.split(':')
+            name, data = pieces[0], pieces[-1]
+            input_datas.update({name: data})
+
+    input_tensors = dict()
+    if args.input_data is not None and atgs.input_shape is not None:
```
Comment (daquexian) [inline onnxsim/__main__.py:62]: `args.input_data is not None` -> `len(input_data_paths) > 0`

`atgs` is a typo

File: onnxsim/onnx_simplifier.py [hunk 3]
```diff
@@ -153,11 +155,16 @@ def is_dynamic(node):
...
-            input_shapes: Optional[TensorShapes] = None) -> Dict[str, np.ndarray]:
+def forward(model,
+            input_data: Optional[Tensors] = None,
+            input_shapes: Optional[TensorShapes] = None,
+            custom_lib: Optional[str] = None) -> Tensors:
     if input_shapes is None:
         input_shapes = {}
     sess_options = rt.SessionOptions()
+    if custom_lib is not None:
+        if os.path.exists(custom_lib):
```
Comment (daquexian) [inline onnxsim/onnx_simplifier.py:166]: raise an exception if file named `custom_lib` does not exist

File: onnxsim/__main__.py [hunk 4]
```diff
@@ -43,12 +49,33 @@ def main():
...
                 name, shape = ':'.join(
                     pieces[:-1]), list(map(int, pieces[-1].split(',')))
-                input_shapes[name] = shape
+                input_shapes.update({name: shape})
+
+    input_datas = dict()
+    if args.input_data is not None:
+        for x in args.input_data:
+            pieces = x.split(':')
+            name, data = pieces[0], pieces[-1]
```
Comment (daquexian) [inline onnxsim/__main__.py:58]: `pieces[0]` -> `:'.join(pieces[:-1])`

**Review by daquexian — COMMENTED** [review]
Thanks for your great contribution! Overall it looks good to me. Some minor comments are posted.

File: onnxsim/onnx_simplifier.py (hunk 3 above)
Comment (tangyanf) [reply onnxsim/onnx_simplifier.py:167]: that's great, i will add it

**Review by daquexian — APPROVED** [review]

**Comment by daquexian** [comment]
Thanks for your great contribution!

**Pull request merged**